### PR TITLE
Feat/auto robot idx

### PR DESCRIPTION
**Pull request by cremebrule**
- Add Stretch Robot
- Auto-infer controller indexing
- Fix controller bugs resulting from recent big env PR

### CODE REVIEW

File: omnigibson/robots/behavior_robot.py [hunk 1]
```diff
@@ -142,10 +142,6 @@ def __init__(
     def usd_path(self):
         return os.path.join(gm.ASSET_PATH, "models/behavior_robot/usd/BehaviorRobot.usd")
 
-    @property
-    def model_name(self):
-        return "BehaviorRobot"
```
Comment (wensi-ai) [inline omnigibson/robots/behavior_robot.py:147]: What's the reason for removing this?

File: omnigibson/robots/stretch.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            ]
+        }
+
+    @property
+    def assisted_grasp_end_points(self):
+        return {
+            self.default_arm: [
+                GraspingPoint(link_name="l_gripper_finger_link", position=[0.025, 0.012, 0.0]),
+                GraspingPoint(link_name="l_gripper_finger_link", position=[-0.025, 0.012, 0.0]),
+            ]
```
Comment (wensi-ai) [inline omnigibson/robots/stretch.py:75]: I'll work on this later today

File: omnigibson/robots/stretch.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+            ["link_gripper_s3_body", "link_aruco_d405"],
+            ["link_gripper_s3_body", "link_gripper_finger_left"],
+            ["link_gripper_finger_left", "link_aruco_fingertip_left"],
+            ["link_gripper_finger_left", "link_gripper_fingertip_left"],
+            ["link_gripper_s3_body", "link_gripper_finger_right"],
+            ["link_gripper_finger_right", "link_aruco_fingertip_right"],
+            ["link_gripper_finger_right", "link_gripper_fingertip_right"],
+            ["respeaker_base", "link_head"],
+            ["respeaker_base", "link_mast"],
+        ]
```
Comment (wensi-ai) [inline omnigibson/robots/stretch.py:129]: Do we actually need to disable this many collisions?

File: omnigibson/objects/controllable_object.py [hunk 4]
```diff
@@ -559,7 +551,7 @@ def get_control_dict(self):
         # TODO: Move gravity force computation dummy to this class instead of BaseRobot
         fcns["gravity_force"] = lambda: (
             ControllableObjectViewAPI.get_generalized_gravity_forces(self.articulation_root_path)
-            if not self.fixed_base
```
Comment (wensi-ai) [inline omnigibson/objects/controllable_object.py:562]: Is this change intended?

File: omnigibson/robots/stretch.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+            self.default_arm: [
+                "joint_lift",
+                "joint_arm_l3",
+                "joint_arm_l2",
+                "joint_arm_l1",
+                "joint_arm_l0",
+                "joint_wrist_yaw",
+                "joint_wrist_pitch",
+                "joint_wrist_roll",
+            ]
```
Comment (ChengshuLi) [inline omnigibson/robots/stretch.py:171]: Why are there more arm links than arm joints? Is it because some of the joints are fixed joints?

File: omnigibson/robots/tiago.py [hunk 6]
```diff
@@ -475,29 +462,16 @@ def base_idx(self):
...
-            "combined": np.array([7, 8, 10, 11, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22]),
-        }
-
-    @property
-    def gripper_control_idx(self):
-        return {"left": np.array([23, 24]), "right": np.array([25, 26])}
+        # Add combined entry
+        idxs = super().arm_control_idx
+        idxs["combined"] = np.sort(np.concatenate([val for val in idxs.values()]))
+        return idxs
```
Comment (ChengshuLi) [inline omnigibson/robots/tiago.py:474]: Is this change intended? What happened to the "left" and the "right"?

File: omnigibson/utils/usd_utils.py [hunk 7]
```diff
@@ -823,7 +827,10 @@ def initialize_view(self):
 
         # Apply the pattern
         expected_prim_paths = {
-            prim_path for prim_path in expected_prim_paths if re.fullmatch(self._pattern.replace("*", ".*"), prim_path)
+            prim_path
+            for prim_path in expected_prim_paths
+            if re.fullmatch(self._pattern.replace("*", ".*"), prim_path)
+            or re.fullmatch(self._pattern.replace("*", ".*").split("/base_link")[0], prim_path)
```
Comment (ChengshuLi) [inline omnigibson/utils/usd_utils.py:833]: hmmm not sure I fully understand this. Also looping in @cgokmen and @hang-yin since I am not too familiar with this API.

**Review by ChengshuLi — APPROVED** [review]

File: omnigibson/robots/behavior_robot.py (hunk 1 above)
Comment (cremebrule) [reply omnigibson/robots/behavior_robot.py:147]: it's automatically generated in the robot base class!

File: omnigibson/robots/stretch.py (hunk 2 above)
Comment (cremebrule) [reply omnigibson/robots/stretch.py:75]: Huge thanks!!

File: omnigibson/robots/stretch.py (hunk 3 above)
Comment (cremebrule) [reply omnigibson/robots/stretch.py:129]: Yeah I manually went through all of them...ideally we should just delete the links themselves, since a lot of the links are very poorly modeled and very trivial, but I was too lazy and wanted to keep it consistent with the raw URDF file LOL

File: omnigibson/objects/controllable_object.py (hunk 4 above)
Comment (cremebrule) [reply omnigibson/objects/controllable_object.py:562]: yes! This was what was causing the falldown issue. we should only be using dummy values if the robot is NOT fixed base

File: omnigibson/robots/fetch.py [hunk 8]
```diff
@@ -455,7 +439,6 @@ def arm_link_names(self):
     def arm_joint_names(self):
         return {
             self.default_arm: [
-                "torso_lift_joint",
```
[the remark this answers is not in the document]
Comment (cremebrule) [reply omnigibson/robots/fetch.py:458]: Yes. This makes it more semantically clear what joint correspond to each part of the robot

File: omnigibson/robots/stretch.py (hunk 5 above)
Comment (cremebrule) [reply omnigibson/robots/stretch.py:171]: Yes, exactly!

File: omnigibson/utils/usd_utils.py (hunk 7 above)
Comment (cremebrule) [reply omnigibson/utils/usd_utils.py:833]: this is a really hacky fix for now, but I believe @cgokmen already has a more general fix for this and for the general /base_link constraint later

File: omnigibson/robots/tiago.py (hunk 6 above)
Comment (cremebrule) [reply omnigibson/robots/tiago.py:474]: It's all automatically inferred now based on arm_joint_names!

File: omnigibson/robots/stretch.py (hunk 2 above)
Comment (wensi-ai) [reply omnigibson/robots/stretch.py:75]: done

**Comment by wensi-ai** [comment]
lgtm. Fixed AG for Stretch
